### PR TITLE
test: Add unit tests for traces endpoint

### PR DESCRIPTION
**Pull request by koxudaxi**
### TL;DR

Added tests for the OpenTelemetry traces endpoint.

### What changed?

Added comprehensive test coverage for the `/api/v1/traces` endpoint that handles OpenTelemetry trace data. The tests verify:

- Successful processing of valid OTLP trace data
- Handling of multiple spans and resource spans
- Processing traces without service name
- Handling empty resource spans
- Proper response structure with partial success fields
- Rejection of invalid trace data
- Error handling for malformed JSON
- Extraction of service name from resource attributes
- Processing of complex attribute values
- OpenAPI documentation for the traces endpoint

### How to test?

Run the test suite with:

```bash
npm test worker/api/traces.test.ts
```

The tests use Vitest and mock console.log to verify the correct logging of trace information.

### Why make this change?

To ensure the OpenTelemetry traces endpoint functions correctly and maintains expected behavior through future changes. These tests validate both the successful processing of valid trace data and proper error handling for edge cases, providing confidence in the reliability of the telemetry collection system.

### CODE REVIEW

**Comment by koxudaxi** [comment]
* **#689** <a href="https://app.graphite.dev/github/pr/Mirascope/lilypad/689?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 2  dependent PRs ([#678](https://github.com/Mirascope/lilypad/pull/678) <a href="https://app.graphite.dev/github/pr/Mirascope/lilypad/678?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#690](https://github.com/Mirascope/lilypad/pull/690) <a href="https://app.graphite.dev/github/pr/Mirascope/lilypad/690?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#688** <a href="https://app.graphite.dev/github/pr/Mirascope/lilypad/688?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#687** <a href="https://app.graphite.dev/github/pr/Mirascope/lilypad/687?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#683** <a href="https://app.graphite.dev/github/pr/Mirascope/lilypad/683?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/Mirascope/lilypad/683?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#682** <a href="https://app.graphite.dev/github/pr/Mirascope/lilypad/682?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `v1`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: worker/api/traces.test.ts
Comment (willbakst) [inline worker/api/traces.test.ts:1]: These tests are a bit of a waste since we'll have to change them the second we actually implement the full `/traces` endpoint. Part of me feels like we should just ignore the dummy endpoint for coverage, but since the tests are already written, I'm happy to just keep as-is and remove later.

File: worker/api/traces.test.ts [hunk 1]
```diff
@@ -0,0 +1,445 @@
+import type { TraceRequest, TraceResponse } from '@/worker/api/traces';
+import app from '@/worker/app';
+import { afterEach, beforeEach, describe, expect, it, vi } from 'vitest';
+
+const testEnv = {
```
Comment (willbakst) [inline worker/api/traces.test.ts]: Don't we already have the environment loaded from `.env.local` from `tests/db/setup.ts`?

Comment (koxudaxi) [reply worker/api/traces.test.ts]: Since it was set to be obtained from Cloud Flare's environment settings, I had prepared a test environment for testing purposes. I will change it to fall back to environment variables locally.

Comment (koxudaxi) [reply worker/api/traces.test.ts]: The environment variables are indeed loaded from .env.local via tests/db/setup.ts, but Hono's `app.request()` doesn't automatically use process.env - it only uses what's
   explicitly passed as the third parameter (the env object).

  The issue is that Hono's context (`c.env`) is completely separate from Node's `process.env`. Even though the environment variables are loaded into process.env by the
  setup file, Hono doesn't automatically use them.

  Without passing an env object to `app.request()`, `c.env` remains undefined in the middleware, causing the "DATABASE_URL is required" error.

  This is why other test files (like worker/app.test.ts) also pass a testEnv object to app.request() - it's a Hono-specific requirement, not a matter of environment 
  variable loading.

Comment (willbakst) [reply worker/api/traces.test.ts]: Got it. Rather than repeating this, why don't we create a new `tests/api/setup.ts` that for now contains only a `TEST_ENV` constant that we can re-use across the various tests, and a middleware that injects it when used? That way we can just use the middleware in each test to ensure the environment is properly passed?

Comment (koxudaxi) [reply worker/api/traces.test.ts]: I just updated the testEnv logic as middleware

File: worker/app.test.ts [hunk 2]
```diff
@@ -1,15 +1,13 @@
...
 import type { HealthResponse } from '@/worker/health/schemas';
 import { describe, expect, it } from 'vitest';
-import app from './app';
+import appBase from './app';
 
-const testEnv = {
-  ENVIRONMENT: 'test',
-  DATABASE_URL: 'postgresql://test@localhost/test',
-};
+const app = createTestApp(appBase);
```
Comment (willbakst) [inline worker/app.test.ts]: Rather than calling `createTestApp` in each test, let's call it a single time in `tests/api/setup.ts` and then import `testApp` from the setup file.

**Review by willbakst — APPROVED** [review]

**Comment by willbakst** [comment]
### Merge activity

* **Sep 10, 3:46 AM UTC**: @willbakst merged this pull request with [Graphite](https://app.graphite.dev/github/pr/Mirascope/lilypad/683).
